File: OCH_Tao/ABC/A/303.py
N = int(input())
S = input()
T = input()
for i in range(N):
  if S[i]==T[i]:
    continue
  elif S[i] in ["1","l"] and T[i] in ["1","l"]:
    continue
  elif S[i] in ["0","o"] and T[i] in ["0","o"]:
    continue
  else:
    print("No")
    break
else:
  print("Yes")

File: OCH_Tao/ABC/A/303.js
const [N, S, T] = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
let flag = true;
for (let i = 0; i < parseInt(N, 10); i++) {
  if (S[i] === T[i]) {
    continue;
  } else if (["1", "l"].includes(S[i]) && ["1", "l"].includes(T[i])) {
    continue;
  } else if (["0", "o"].includes(S[i]) && ["0", "o"].includes(T[i])) {
    continue;
  } else {
    console.log("No");
    flag = false;
    break;
  }
}
if (flag) {
  console.log("Yes");
}